core/trino-main/src/main/java/io/trino/operator/join/unspilled/HashBuilderOperator.java

```diff
@@ -367,78 +275,17 @@ private void updateIndex(Page page)
     {
         index.addPage(page);
 
-        if (spillEnabled) {
-            localRevocableMemoryContext.setBytes(index.getEstimatedSize().toBytes());
-        }
-        else {
-            if (!localUserMemoryContext.trySetBytes(index.getEstimatedSize().toBytes())) {
-                index.compact();
-                localUserMemoryContext.setBytes(index.getEstimatedSize().toBytes());
-            }
+        if (!localUserMemoryContext.trySetBytes(index.getEstimatedSize().toBytes())) {
+            index.compact();
+            localUserMemoryContext.setBytes(index.getEstimatedSize().toBytes());
         }
         operatorContext.recordOutput(page.getSizeInBytes(), page.getPositionCount());
     }
 
-    private void spillInput(Page page)
-    {
-        checkState(spillInProgress.isDone(), "Previous spill still in progress");
-        checkSuccess(spillInProgress, "spilling failed");
-        spillInProgress = getSpiller().spill(page);
-    }
-
     @Override
     public ListenableFuture<Void> startMemoryRevoke()
```

Comment:
remove this method, there is already a check state somewhere in `OperatorContext` that validates that spill is not triggered on method with no revocable mem